### PR TITLE
api: Sync update-stream event types with doc and align implementation

### PR DESCRIPTION
**Pull request by chrisbobbe**
After #5341, on the path to fixing #5250.

### CODE REVIEW

File: src/api/eventTypes.js [hunk 1]
```diff
@@ -149,21 +149,91 @@ type StreamListEvent = $ReadOnly<{|
...
+  | {|
+      ...StreamUpdateEventBase,
+      +property: 'rendered_description',
+      +value: $PropertyType<Stream, 'rendered_description'>,
+    |}
+  | {|
+      ...StreamUpdateEventBase,
+      +property: 'is_web_public',
+      +value: $PropertyType<Stream, 'is_web_public'>,
+    |}
```
Comment (gnprice) [inline src/api/eventTypes.js:193]: I think these can usefully be further deduplicated. I'll send that as a quick PR on top of this.

Comment (gnprice) [reply src/api/eventTypes.js:193]: #5350

File: src/streams/__tests__/streamsReducer-test.js [hunk 2]
```diff
@@ -25,23 +21,20 @@ describe('streamsReducer', () => {
...
-
-      const expectedState = [stream1, stream2];
-
-      const newState = streamsReducer(initialState, action);
-
-      expect(newState).toEqual(expectedState);
+      expect(
+        streamsReducer(
+          [],
+          deepFreeze({
```
[the remark this answers is not in the document]
Comment (chrisbobbe) [reply src/streams/__tests__/streamsReducer-test.js:27]: I've just tried this out, and I haven't yet found a way to do this without Flow errors; I think a factor is how many branches there are in `StreamEvent`.

[the remark this answers is not in the document]
Comment (chrisbobbe) [reply src/streams/__tests__/streamsReducer-test.js:27]: Maybe I'll come back to it in the future when doing other substantive work in this file.

[the remark this answers is not in the document]
Comment (chrisbobbe) [reply src/streams/__tests__/streamsReducer-test.js:27]: Just sent #5352 for the first thing though.

[the remark this answers is not in the document]
Comment (gnprice) [reply src/streams/__tests__/streamsReducer-test.js:27]: > I've just tried this out, and I haven't yet found a way to do this without Flow errors; I think a factor is how many branches there are in `StreamEvent`.

Ah indeed. I think I have a solution for that; I'll send as a PR after merging #5352.

[the remark this answers is not in the document]
Comment (gnprice) [reply src/streams/__tests__/streamsReducer-test.js:27]: OK, sent: #5353. The extra bit it needs is a type annotation on the helper, making it generic with a bound:
```js
const mkAction = <E: $Diff<StreamEvent, {| id: mixed, type: mixed |}>>(event: E) =>
  deepFreeze({ type: EVENT, event: { id: 0, type: EventTypes.stream, ...event } });
```

Wrote a bit of a long commit message for it, too -- I tried to think through why it is that this was getting errors without an annotation, and why this particular spot is one where Flow needs an annotation even though in many other situations it infers things just fine, and then to see if I could explain that in a meaningful way.

**Pull request merged**